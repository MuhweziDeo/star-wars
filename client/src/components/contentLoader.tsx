import {FlexGrid, FlexGridItem} from "baseui/flex-grid";
import {Skeleton} from "baseui/skeleton";
import React from "react";

export interface Props {
    contentLength?: number
}
export const ContentLoader: React.FunctionComponent<Props> = ({contentLength = 15}) => {
    const renderSkeleton = () => Array.from(Array(contentLength).keys()).map((index) => <FlexGridItem key={index}><Skeleton height="150px" width="300px"/></FlexGridItem> )
    return (
        <FlexGrid
            className="loader"
            data-testid = "loader"
            flexGridColumnCount={4}
            flexGridColumnGap="scale800"
            flexGridRowGap="scale800"
        >
            {renderSkeleton()}

        </FlexGrid>
    )
}