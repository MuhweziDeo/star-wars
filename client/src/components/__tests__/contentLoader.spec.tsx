import React from 'react';
import { FlexGrid } from 'baseui/flex-grid';
import TestRenderer from 'react-test-renderer';
import { ContentLoader } from '../contentLoader';

describe("ContentLoader", () => {
    it("should render without crushing", () => {
        const component = TestRenderer.create(<ContentLoader/>)
        expect(component.toJSON()).toMatchSnapshot();
    })

    it("should render 15 items by default", () => {
        const component = TestRenderer.create(<ContentLoader/>)
        expect(component.toJSON()).toMatchSnapshot();
        const flexGrid = component.root.findByType(FlexGrid)
        expect(flexGrid.props.children.length).toBe(15);
    })

    it("should render number items by provided contentLength", () => {
        const component = TestRenderer.create(<ContentLoader contentLength={30}/>)
        expect(component.toJSON()).toMatchSnapshot();
        const flexGrid = component.root.findByType(FlexGrid)
        expect(flexGrid.props.children.length).toBe(30);
    })
})