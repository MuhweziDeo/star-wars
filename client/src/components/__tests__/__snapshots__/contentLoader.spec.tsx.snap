// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentLoader should render 15 items by default 1`] = `
<div
  className="loader "
  data-baseweb="flex-grid"
  data-testid="loader"
>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
</div>
`;

exports[`ContentLoader should render 15 items by default 2`] = `
<div
  className="loader "
  data-baseweb="flex-grid"
  data-testid="loader"
>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
</div>
`;

exports[`ContentLoader should render items by default 1`] = `
<div
  className="loader "
  data-baseweb="flex-grid"
  data-testid="loader"
>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
</div>
`;

exports[`ContentLoader should render number items by provided contentLen 1`] = `
<div
  className="loader "
  data-baseweb="flex-grid"
  data-testid="loader"
>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
</div>
`;

exports[`ContentLoader should render number items by provided contentLength 1`] = `
<div
  className="loader "
  data-baseweb="flex-grid"
  data-testid="loader"
>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
</div>
`;

exports[`ContentLoader should render number items by provided num 1`] = `
<div
  className="loader "
  data-baseweb="flex-grid"
  data-testid="loader"
>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
</div>
`;

exports[`ContentLoader should render number items by w 1`] = `
<div
  className="loader "
  data-baseweb="flex-grid"
  data-testid="loader"
>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
</div>
`;

exports[`ContentLoader should render without crushing 1`] = `
<div
  className="loader "
  data-baseweb="flex-grid"
  data-testid="loader"
>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
</div>
`;

exports[`ContentLoader should render without crushing 2`] = `
<div
  className="loader "
  data-baseweb="flex-grid"
  data-testid="loader"
>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
  <div
    className=""
    data-baseweb="flex-grid-item"
  >
    <div
      className=""
      testid="loader"
    />
  </div>
</div>
`;
